File: app/src/main/java/com/victoria/bleled/data/remote/myservice/IMyRemoteService.java
package com.victoria.bleled.data.remote.myservice;

import android.util.Log;

import androidx.lifecycle.LiveData;

import com.google.gson.Gson;
import com.victoria.bleled.data.model.ModelAppInfo;
import com.victoria.bleled.data.model.ModelUpload;
import com.victoria.bleled.util.arch.network.NetworkResult;
import com.victoria.bleled.util.arch.network.RetrofitHelper;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.List;

import okhttp3.MultipartBody;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;

public interface IMyRemoteService {
    /************************************************************
     *  Constants
     ************************************************************/
    String API_BASE_URL = "http://192.168.0.13:9101/api/";
    String API_KEY = "clubonline20200519key";

    /************************************************************
     *  Functions
     ************************************************************/
    static String encrypt(String... params) {
        HashMap<String, String> map = new HashMap<>();
        for (int i = 0; i < params.length; i += 2) {
            map.put(params[i], params[(i + 1)]);
        }
        String r = RetrofitHelper.encryptParams(IMyRemoteService.API_KEY, map);
        return r;
    }

    static String encrypt(HashMap<String, String> map) {
        String r = RetrofitHelper.encryptParams(IMyRemoteService.API_KEY, map);
        return r;
    }

    //  BaseResponse<ModelAppInfo> baseResponse = decrypt(new TypeToken<BaseResponse<ModelAppInfo>>() {}.getType(), response.data);
    static <T> BaseResponse<T> decrypt(Type respType, String response) {
        String strJson = RetrofitHelper.decrypt(IMyRemoteService.API_KEY, response);
        BaseResponse<T> baseResponse = new Gson().fromJson(strJson, respType);
        Log.d("IShequService", strJson);
        return baseResponse;
    }


    /************************************************************
     *  Common
     ************************************************************/
    @FormUrlEncoded
    @POST("common/app_info")
    LiveData<NetworkResult<BaseResponse<ModelAppInfo>>> appInfo(@Field("dev_type") String dev_type);

    @Multipart
    @POST("common/upload_file")
    LiveData<NetworkResult<BaseResponse<ModelUpload>>> upload(@Part MultipartBody.Part file);

    @Multipart
    @POST("common/multi_upload_file")
    LiveData<NetworkResult<BaseResponse<List<ModelUpload>>>> upload(@Part List<MultipartBody.Part> files);
}

File: app/src/main/java/com/victoria/bleled/data/remote/myservice/IMyRemoteService.kt
package com.victoria.bleled.data.remote.myservice

import android.util.Log
import androidx.lifecycle.LiveData
import com.google.gson.Gson
import com.victoria.bleled.data.model.ModelAppInfo
import com.victoria.bleled.data.model.ModelUpload
import com.victoria.bleled.util.arch.network.NetworkResult
import com.victoria.bleled.util.arch.network.RetrofitHelper
import okhttp3.MultipartBody.Part
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.Multipart
import retrofit2.http.POST
import java.lang.reflect.Type
import java.util.*

interface IMyRemoteService {
    /************************************************************
     * Common
     */
    @FormUrlEncoded
    @POST("common/app_info")
    fun appInfo(@Field("dev_type") dev_type: String): LiveData<NetworkResult<BaseResponse<ModelAppInfo>>>

    @Multipart
    @POST("common/upload_file")
    fun upload(@retrofit2.http.Part file: Part): LiveData<NetworkResult<BaseResponse<ModelUpload>>>

    @Multipart
    @POST("common/multi_upload_file")
    fun uploadList(@retrofit2.http.Part files: List<Part>): LiveData<NetworkResult<BaseResponse<List<ModelUpload>>>>

    companion object {
        /************************************************************
         * Functions
         */
        fun encrypt(vararg params: String): String {
            val map = HashMap<String, String>()
            var i = 0
            while (i < params.size) {
                map[params[i]] = params[i + 1]
                i += 2
            }
            return RetrofitHelper.encryptParams(API_KEY, map)
        }

        fun encrypt(map: HashMap<String, String>): String {
            return RetrofitHelper.encryptParams(API_KEY, map)
        }

        //  BaseResponse<ModelAppInfo> baseResponse = decrypt(new TypeToken<BaseResponse<ModelAppInfo>>() {}.getType(), response.data);
        fun <T> decrypt(respType: Type, response: String): BaseResponse<T> {
            val strJson = RetrofitHelper.decrypt(API_KEY, response)
            val baseResponse: BaseResponse<T> = Gson().fromJson(strJson, respType)
            Log.d("IShequService", strJson)
            return baseResponse
        }

        /************************************************************
         * Constants
         */
        const val API_BASE_URL = "http://192.168.0.13:9101/api/"
        const val API_KEY = "clubonline20200519key"
    }
}